common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -25,6 +25,8 @@
 import com.google.common.collect.Lists;
 import org.apache.hadoop.conf.Configuration;
 import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.util.DiskChecker;
```

Comment:
The problem is when using default settings(recovery and spark.yarn.shuffle.stopOnFailure were disabled), if localDirs[0] was broken, then NodeManager can continue work, but spark tasks running on that NodeManager will fail due to can't connected to port 7337.